navigation.py

```diff
@@ -0,0 +1,71 @@
+# Navigation module for ThatsJustTheWazeItIs
+from collections import defaultdict
+import json
+import threading
+from map import Graph
+
+class Navigation:
+    """docstring for ."""
+
+    def __init__(self, filename):
+        self.graph = Graph(filename)
+        self.__roomEmpty = threading.Lock()
+        self.__turnstile = threading.Lock()
+
+    def requestMapUpdate(self, body):
+        self.__turnstile.acquire()
+        self.__turnstile.release()
+
+        with self.__roomEmpty:
+            path, wait, total_wait = self.dijkstra(body["curr"], body["dest"])
+        
+        print("PATH")
+        print(path)
+    
+        if len(path) == 1:
+            road = ""
```

Comment:
If the path is length 1, do we still need to traverse the edge (street) between the car's current node and the destination? Like if we're at node A and Packard Ave is between A and destination node B, I think you'd still want to tell the car to drive down Packard Ave.